server/channels/app/channel.go

```diff
@@ -1360,15 +1360,62 @@ func (a *App) UpdateChannelMemberNotifyProps(c request.CTX, data map[string]stri
 	a.invalidateCacheForChannelMembersNotifyProps(member.ChannelId)
 
 	// Notify the clients that the member notify props changed
+	err = a.sendUpdateChannelMemberNotifyPropsEvent(member)
+	if err != nil {
+		return nil, model.NewAppError("UpdateChannelMemberNotifyProps", "api.marshal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	return member, nil
+}
+
+func (a *App) UpdateChannelMembersNotifyProps(c request.CTX, members []*model.ChannelMember) ([]*model.ChannelMember, *model.AppError) {
+	updated, err := a.Srv().Store().Channel().UpdateMultipleMembersNotifyProps(members)
+	if err != nil {
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &appErr):
+			return nil, appErr
+		default:
+			return nil, model.NewAppError("UpdateMultipleMembersNotifyProps", "app.channel.update_channel_members_notify_props.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+
+	// Invalidate caches for the users and channels that have changed
+	userIds := make(map[string]bool)
+	channelIds := make(map[string]bool)
+	for _, member := range updated {
+		userIds[member.UserId] = true
+		channelIds[member.ChannelId] = true
+	}
+
+	for userId := range userIds {
+		a.InvalidateCacheForUser(userId)
+	}
+	for channelId := range channelIds {
+		a.invalidateCacheForChannelMembersNotifyProps(channelId)
+	}
```

Comment:
We can combine this in a single loop:

```go
userIds := make(map[string]bool)
channelIds := make(map[string]bool)
for _, member := range updated {
  if !userIds[member.UserId] {
  userIds[member.UserId] = true
  a.InvalidateCacheForUser(member.UserId)
}

  if !channelIds[member.ChannelId] {
  channelIds[member.ChannelId] = true
  a.invalidateCacheForChannelMembersNotifyProps(channelId)
 }
}

```

Reply:
Would using a single loop have a performance benefit here? If it's more about code style, I actually find it easier to follow what the code is doing by having it broken up into two steps of making a "set" of IDs and then invalidating them

Reply:
Well, yes it would have a perf benefit - you would replace 3 loops with 1. Personally, I don't find this to be any harder to read than the separate loops solution. But not a big deal. I'll defer to you.

Reply:
In that case, I'm going to leave it as-is then since I think the extra overhead of the loops is probably negligible compared to the cost of invalidating the caches